lib/linked-list.js

```diff
@@ -125,42 +200,118 @@ class LinkedList {
     /*
     method to reverse the singly linked list
     note: the nodes should be moved and not just the values in the nodes
-    Time Complexity: ?
-    Space Complexity: ?
+    Time Complexity: O(n)
+    Space Complexity: O(1)
     */
+   // 2 -> 3 -> 4 -> 5
     reverse() {
-        throw new Error("This method hasn't been implemented yet!");
+        if (!this.#head) {
+            return;
+        };
+        
+        let previous = null;
+        let current = this.#head;
+        let next = null;
+
+        while (current) {
+            next = current.next;
+            current.next = previous;
+            previous = current;     
+            current = next;     
+        }
+        this.#head = previous;
     }
 
     // Advanced Exercises
     /*
     returns the value at the middle element in the singly linked list
-    Time Complexity: ?
-    Space Complexity: ?
+    Time Complexity: O(n)
+    Space Complexity: O(1)
      */
     findMiddleValue() {
```

Comment:
👍 You could also have reused the `length` method as well as the get by index method